lib/charms/observability_libs/v1/kubernetes_service_patch.py

```diff
@@ -3,14 +3,11 @@
 
 """# KubernetesServicePatch Library.
 
-This library is designed to enable developers to more simply patch the Kubernetes Service created
-by Juju during the deployment of a sidecar charm. When sidecar charms are deployed, Juju creates a
-service named after the application in the namespace (named after the Juju model). This service by
-default contains a "placeholder" port, which is 65535/TCP.
+[DEPRECATED!]
+The `kubernetes_service_patch` library is DEPRECATED for patching the Kubernetes service created
+by Juju during the deployment of a charm and `ops.Unit.set_ports` functionality should be used for that instead.
```

Comment:
After discussion with @simskij and @IbraAoad, We think that we should deprecate the entire library, not just the part pertaining to juju owned resources.

Reply:
Also, we should add to the deprecation a removal date of October 2025.